jobs/github/scripts/snippet_build_basic-make-check.sh

```diff
@@ -2,11 +2,18 @@
 
     NEEDS_MGO=$(echo "${NEEDS_MGO}" | tr '[:upper:]' '[:lower:]')
     if [ "${NEEDS_MGO}" = "true" ]; then
-      echo "Installing juju-db snap."
+      action=$(snap info juju-db | grep -q "installed" || echo "install")
+      if [ "$action" == "" ]; then
+              action="refresh"
+      fi
+
       # when running inside a privileged container, snapd fails because udevd isn't
       # running, but on the second occurance it is.
       # see: https://github.com/lxc/lxd/issues/4308
-      sudo snap refresh juju-db --channel=4.4 2> /dev/null || sudo snap install juju-db --channel=4.4
+      for i in {1..3}; do
+        echo $action"ing juju-db snap. Attempt ""$i"
```

Comment:
Interesting choice of syntax. I probably would have written:
```suggestion
        echo "${action}ing juju-db snap. Attempt ${i}"
```

(I note that you didn't realize $ gets expanded inside double quotes as your commit message is "Use instead of unconditionally install" and your `$action` got expanded by the shell into the empty string.

Reply:
Yeah my string formatting in bash is not the greatest!

> (I note that you didn't realize $ gets expanded inside double quotes as your commit message is "Use instead of unconditionally install" and your $action got expanded by the shell into the empty string.

Also this :laughing:  :point_up: